### PR TITLE
Additional small clean-ups from openshift/installer review

### PR DESCRIPTION
**Pull request by stbenjam**
See commits for details, but this removes FIXME's and fixes some ordering issues.

### CODE REVIEW

**Comment by stbenjam** [comment]
Don't merge until latest rebase is in, to avoid creating conflicts.

File: pkg/asset/rhcos/image.go [hunk 1]
```diff
@@ -67,7 +67,6 @@ func (i *Image) Generate(p asset.Parents) error {
 		//TODO(serbrech): change to right image once available.
 		osimage = "/resourceGroups/rhcos_images/providers/Microsoft.Compute/images/rhcostestimage"
 	case baremetal.Name:
-		// FIXME: baremetal
 		osimage, err = rhcos.QEMU(ctx)
```
Comment (stbenjam) [inline pkg/asset/rhcos/image.go:72]: This doesn't need fixing, from @russellb 

```
using the same image as libvirt is correct for the bootstrap VM
we do have: https://github.com/openshift-metal3/kni-installer/issues/37
which covers "use the right image for masters and workers"
```

File: pkg/asset/machines/master.go [hunk 2]
```diff
@@ -158,7 +158,6 @@ func (m *Master) Generate(dependencies asset.Parents) error {
 		}
 		azure.ConfigMasters(machines, clusterID.InfraID)
 	case baremetaltypes.Name:
-		// FIXME: baremetal
```
Comment (stbenjam) [inline pkg/asset/machines/master.go:182]: This was introduced in a rebase,  before we created machines for baremetal.  We create them now, so this FIXME (and the one below,I think) aren't needed.

File: pkg/asset/rhcos/image.go (hunk 1 above)
Comment (hardys) [reply pkg/asset/rhcos/image.go:72]: Yeah also see https://github.com/openshift-metal3/kni-installer/pull/100 where we need to consume two different images which fixes that issue, so I bypassed this abstraction - I think removing this FIXME is reasonable.

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/898/

**Comment by metal3ci** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/900/

File: pkg/asset/machines/master.go (hunk 2 above)
Comment (markmc) [reply pkg/asset/machines/master.go:182]: The thing that bothers me now with this code is that we're not using rhcosImage - i.e. nowhere in our machines resource are we describing the bootimage we used, unlike other platforms (except libvirt)

Comment (markmc) [reply pkg/asset/machines/master.go:182]: I've filed openshift-metal3/kni-installer#160 now

Even though the baremetal code path doesn't use the rhcosImage variable right now, it bothers me that the variable contains the wrong data for masters ...

File: pkg/asset/rhcos/image.go (hunk 1 above)
Comment (markmc) [reply pkg/asset/rhcos/image.go:72]: This is ... ugh ...

Some of the underlying assumptions of the `rhcos.Image` abstraction:

1. The same image is used for bootstrap, masters, and workers
2. That image information is included in the generated machines manifests and also to terraform

I think we should follow that basic idea and just have an "oh hey, our bootstrap node boot image is different!" special case

**Review by markmc — APPROVED** [review]
Happy to remove the FIXMEs now that openshift-metal3/kni-installer#160 tracks them and gives more detail

File: pkg/asset/rhcos/image.go (hunk 1 above)
Comment (hardys) [reply pkg/asset/rhcos/image.go:72]: Yeah agree we need a better abstraction here which caters for the situation where the bootstrap and master images are different, I hacked around it in #100 but would welcome feedback on possible approaches there.

**Pull request merged**